CMakeLists.txt

```diff
@@ -198,116 +198,105 @@ if(BUILD_SHARED_LIBS)
     endif()
 endif()
 
+add_library(samplerate_config INTERFACE)
+# To allow src_config.h and config.h to be included
+target_include_directories(samplerate_config INTERFACE src ${PROJECT_BINARY_DIR})
+add_library(samplerate::config ALIAS samplerate_config)
```

Comment:
No better than `target_include_directories`.  Entities should not be multiplied without necessity.

Reply:
IMO it is better: If you compare it to a programming language this is the equivalent of a named function. So instead of shoving some folders in there (which BTW needs an extra command while you already have a target_link_libraries) you link to a named entity, in this case one meant to give you access to the configuration. The comment makes the intention clear, repeating that on every use would be noisy. And if that folder ever need to change there is a single place for it. So again this can be though of as `string getConfigIncludeFolder(){ ... }`
-> Diff shows that: 13 LOC less after this change which IMO is better for maintainability.

Reply:
> IMO it is better: If you compare it to a programming language this is the equivalent of a named function. So instead of shoving some folders in there (which BTW needs an extra command while you already have a target_link_libraries) you link to a named entity, in this case one meant to give you access to the configuration. The comment makes the intention clear, repeating that on every use would be noisy. And if that folder ever need to change there is a single place for it. So again this can be though of as string getConfigIncludeFolder(){ ... }

Do not forget that other people will also look into the project, who are not particularly interested in understanding what kind of library it is and why it is needed. On the other hand, target_include_directories() is clear to anyone, even with a basic knowledge of CMake. So my verdict is no.

> -> Diff shows that: 13 LOC less after this change which IMO is better for maintainability.

Compact code is also not particularly important, we are not writing in Assembler here.